### ISSUE TITLE
Match.Where being instantiated as a class, causes error on Meteor 1.6.1 boot

### ISSUE THREAD

**Issue by jasonschock** [aldeed/meteor-simple-schema#722]
Please see issue meteor/meteor#9383 for context and specific https://github.com/meteor/meteor/issues/9383#issuecomment-345675174.

Basically, this error is now appearing with Meteor 1.6.1 beta on boot:

```
W20171128-13:29:39.919(-8)? (STDERR) /Users/jschock/.meteor/packages/meteor-tool/.1.6.1-beta.11.js08xa.q30b++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/server-lib/node_modules/fibers/future.js:280
W20171128-13:29:39.920(-8)? (STDERR) 						throw(ex);
W20171128-13:29:39.921(-8)? (STDERR) 						^
W20171128-13:29:39.921(-8)? (STDERR)
W20171128-13:29:39.921(-8)? (STDERR) TypeError: Match.Where is not a constructor
W20171128-13:29:39.922(-8)? (STDERR)     at packages/aldeed_simple-schema.js:1547:26
W20171128-13:29:39.922(-8)? (STDERR)     at packages/aldeed_simple-schema.js:2160:4
W20171128-13:29:39.922(-8)? (STDERR)     at packages/aldeed_simple-schema.js:3128:3
W20171128-13:29:39.922(-8)? (STDERR)     at infos.forEach.info (/Users/jschock/Code/calliope/calliope/.meteor/local/build/programs/server/boot.js:415:13)
W20171128-13:29:39.923(-8)? (STDERR)     at Array.forEach (<anonymous>)
W20171128-13:29:39.924(-8)? (STDERR)     at /Users/jschock/Code/calliope/calliope/.meteor/local/build/programs/server/boot.js:414:9
W20171128-13:29:39.924(-8)? (STDERR)     at /Users/jschock/Code/calliope/calliope/.meteor/local/build/programs/server/boot.js:468:5
W20171128-13:29:39.924(-8)? (STDERR)     at Function.run (/Users/jschock/Code/calliope/calliope/.meteor/local/build/programs/server/profile.js:510:12)
W20171128-13:29:39.924(-8)? (STDERR)     at /Users/jschock/Code/calliope/calliope/.meteor/local/build/programs/server/boot.js:467:11
```

Per comments by @abernix:

> It seems aldeed:simple-schema could accomplish the same thing by not instantiating a new Match.Where as if it were a class.

**Comment by benjamn** [comment]
It seems as though switching `Match.Where` to method shorthand syntax in [this commit](https://github.com/meteor/meteor/commit/dd336553751e83cbe00f7266f98e007f0fd00fe3) caused it to no longer be usable as a constructor. Instead, `Match.Where` should now be called as a function, without `new`. Invoking it as a constructor was creating an unnecessary object before, so this change would be a good idea even if it wasn't strictly necessary.

**Comment by skirunman** [comment]
Seeing the same problem with aldeed:simple-schema@1.5.3 running METEOR@1.6.1-beta.18

```
=> Exited with code: 1
W20171213-11:42:54.253(-8)? (STDERR) /Users/jim/.meteor/packages/meteor-tool/.1.6.1-beta.18.1cdl7le++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle/server-lib/node_modules/fibers/future.js:280
W20171213-11:42:54.253(-8)? (STDERR) 						throw(ex);
W20171213-11:42:54.253(-8)? (STDERR) 						^
W20171213-11:42:54.254(-8)? (STDERR) 
W20171213-11:42:54.254(-8)? (STDERR) TypeError: Match.Where is not a constructor
W20171213-11:42:54.254(-8)? (STDERR)     at packages/aldeed_simple-schema.js:1547:26
W20171213-11:42:54.255(-8)? (STDERR)     at packages/aldeed_simple-schema.js:2160:4
W20171213-11:42:54.255(-8)? (STDERR)     at packages/aldeed_simple-schema.js:3128:3
.
.
.
```

@aldeed Any chance PR #723 gets merged and new version published to Atmosphere or should we just fork?

**Comment by crapthings** [comment]
upgrade  to 1.6.1 beta too, same issue here.

**Comment by skirunman** [comment]
If you don't want to create your own fork you can install ours as a local package. 
https://github.com/clozertech/meteor-simple-schema
We also merged #616. Works great with Meteor 1.6.1.

**Comment by aldeed** [comment]
The Meteor SimpleSchema package is no longer maintained other than critical fixes to keep it running with each latest Meteor release. Closing non-critical issues in this repo. Anyone who cares about this may do one or more of the following:

- Switch to the [NPM package](https://github.com/aldeed/simple-schema-js). Be sure to adjust for the [breaking changes](https://github.com/aldeed/meteor-simple-schema/blob/master/CHANGELOG.md#200). There can be a lot of work involved in switching, but it is worth the effort.
- If this issue still occurs in the NPM package, [submit an issue there](https://github.com/aldeed/simple-schema-js/issues). Be sure to include a link to a reproduction repo or a PR that adds a failing test for the issue.
- If you have more time than money, submit a pull request to resolve any [issue labeled "help wanted"](https://github.com/aldeed/simple-schema-js/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) in the NPM package repo.
- If you have more money than time, [donate](https://opencollective.com/simple-schema-js/donate) to help support development if you are able
